src/controls.c

```diff
@@ -0,0 +1,100 @@
+#include "../include/controls.h"
+#include "SDL.h"
+#include <stdio.h>
+
+// Function definitions
+void init_emulator(State *state)
+{
+    // Initialize all necessary parts of the state
+    state->a = 0;
+    state->pc = 0;
+    state->input_state = 0;
+    // Initialize other parts of the state as needed
+}
+
+// Emulate 8080 processor (stub function)
+void emulate8080(State *state)
+{
+    // Implement the 8080 emulation logic here
+}
+
+// Update input state based on SDL events
+void update_input_state(State *state, SDL_Event *e)
+{
+    // Handle key press events
+    if (e->type == SDL_KEYDOWN)
+    {
+        switch (e->key.keysym.sym)
+        {
+        case SDLK_LEFT:
+        {
+            state->input_state |= (1 << 5); // P1 joystick left
```

Comment:
I don't think we're going to have a "state->input_state". I think you will want to set the appropriate port (either 1 or 2) so the program can read it from the register
http://www.emulator101.com/buttons-and-ports.html